File: test/javascript/getApi.js
const assert = require('assert');
var EndoflifeDate = require('endoflife-date');

describe('getApi', function() {

  // Commented out for now due to issue

  describe('getApiAllJson', function() {
    it('should respond without error', function(done) {
      var callback = function(error, data, response) {
        if (error) {
          console.error(error);
        } else {
          assert.equal(error, undefined);
          assert.equal(typeof data, 'object');
          // console.dir(data);
          assert.equal(Array.isArray(data), true);
          assert.equal(data.includes('nodejs'), true);
          assert.equal(data.includes('python'), true);
          assert.equal(data.includes('ruby'), true);
          assert.equal(response.res.statusCode, 200);
        }
        done(error);
      };
      const api = new EndoflifeDate.DefaultApi();
      api.getApiAllJson(callback);
    });
  });

  describe('getApiProductCycleJson', function() {
    it('should respond without error', function(done) {
      done();
      // Disabled due to https://github.com/oapicf/endoflife.date-api-clients/issues/1
      // var callback = function(error, data, response) {
      //   if (error) {
      //     console.error(error);
      //   } else {
      //     assert.equal(error, undefined);
      //     assert.equal(typeof data, 'object');
      //     console.dir(data);
      //     assert.equal(data.eol, '2028-10-31');
      //     assert.equal(response.res.statusCode, 200);
      //   }
      //   done(error);
      // };
      // const api = new EndoflifeDate.DefaultApi();
      // api.getApiProductCycleJson('python', '3.12', callback);
    });
  });

  describe('getApiProductJson', function() {
    it('should respond without error', function(done) {
      done();
      // Disabled due to https://github.com/oapicf/endoflife.date-api-clients/issues/1
      // var callback = function(error, data, response) {
      //   if (error) {
      //     console.error(error);
      //     done(error);
      //   } else {
      //     assert.equal(error, undefined);
      //     assert.equal(typeof data, 'object');
      //     console.dir(data);
      //     assert.equal(Array.isArray(data.cycles), true);
      //     let found = false;
      //     for (let i = 0; i < data.cycles.length; i++) {
      //       if (data.cycles[i].cycle === '3.12') {
      //         found = true;
      //         break;
      //       }
      //     }
      //     assert.equal(found, true);
      //     assert.equal(response.res.statusCode, 200);
      //   }
      //   done(error);
      // };
      // const api = new EndoflifeDate.DefaultApi();
      // api.getApiProductJson('python', callback);
    });
  });

});

File: test/python/get_api.py
import unittest
import endoflifedate
from endoflifedate.rest import ApiException
from pprint import pprint

class TestEndOfLifeDateGetApi(unittest.TestCase):

    def test_get_api_all_json(self):

      configuration = endoflifedate.Configuration(
        host = "https://endoflife.date"
      )

      with endoflifedate.ApiClient(configuration) as api_client:

          api_instance = endoflifedate.DefaultApi(api_client)

          try:
              api_response = api_instance.get_api_all_json()
              print("The response of DefaultApi->get_api_all_json:\n")
              # pprint(api_response)
              all_products = api_response
              assert 'nodejs' in all_products
              assert 'python' in all_products
              assert 'ruby' in all_products

          except ApiException as e:
              self.fail('Exception when calling DefaultApi->get_api_all_json: %s\n' % e)

    def test_get_api_product_cycle_json(self):

      configuration = endoflifedate.Configuration(
        host = "https://endoflife.date"
      )

      with endoflifedate.ApiClient(configuration) as api_client:

          api_instance = endoflifedate.DefaultApi(api_client)

          try:
              api_response = api_instance.get_api_product_cycle_json('python', '3.12')
              print("The response of DefaultApi->get_api_product_cycle_json:\n")
              # pprint(api_response)
              product_cycle = api_response
              assert product_cycle.eol.to_dict() == '2028-10-31'

          except ApiException as e:
              self.fail('Exception when calling DefaultApi->get_api_all_json: %s\n' % e)

    def test_get_api_product_json(self):

      configuration = endoflifedate.Configuration(
        host = "https://endoflife.date"
      )

      with endoflifedate.ApiClient(configuration) as api_client:

          api_instance = endoflifedate.DefaultApi(api_client)

          try:
              api_response = api_instance.get_api_product_json('python')
              print("The response of DefaultApi->get_api_product_json:\n")
              # pprint(api_response)
              product_all_cycles = api_response
              python_3_12_cycle = None
              for cycle in product_all_cycles:
                # print(cycle.cycle)
                if cycle.cycle.to_dict() == '3.12':
                  python_3_12_cycle = cycle
                  break
              if python_3_12_cycle is None:
                self.fail('3.12 cycle not found in python product cycles')
              assert python_3_12_cycle.eol.to_dict() == '2028-10-31'

          except ApiException as e:
              self.fail('Exception when calling DefaultApi->get_api_all_json: %s\n' % e)
